linux_boot_info/src/lib.rs

```diff
@@ -0,0 +1,626 @@
+    /// In use or reserved by the system.
+    RESERVED = 2,
+    /// ACPI Reclaim Memory. Available after the OS reads the ACPI tables.
+    ACPI = 3,
+    /// ACPI NVS memory; in use or reserved by the system.
+    NVS = 4,
+    /// Memory in which errors have been detected.
+    UNUSABLE = 5,
+    /// Memory that is not enabled.
+    DISABLED = 6,
+    /// Persistent memory: must be handled distinct from conventional volatile memory.
+    PMEM = 7,
+}
+
+bitflags! {
+    /// Boot protocol option flags.
+    pub struct LoadFlags: u8 {
+        /// - If `0`, the protected-mode code is loaded at `0x10000`.
+        /// - If `1`, the protected-mode code is loaded at `0x100000`.
+        const LOADED_HIGH = 1 << 0;
+        /// Used internally by the compressed kernel to communicate KASLR status to kernel proper.
+        ///
+        /// - If `1`, KASLR enabled.
+        /// - If `0`, KASLR disabled.
+        const KASLR_FLAG = 1 << 1;
+        /// Requests the kernel to not write early messages that require accessing the display hardware directly.
+        ///
+        ///  - If 0, print early messages.
+        ///  - If 1, suppress early messages.
+        const QUIET_FLAG = 1 << 5;
+        #[deprecated]
+        const KEEP_SEGMENTS = 1 << 6;
+        /// Indicates that the value entered in [`SetupHeader::heap_end_ptr`] is valid.
+        ///
+        /// If this field is clear, some setup code functionality will be disabled.
+        const CAN_USE_HEAP = 1 << 7;
+    }
+
+    /// Extended Boot protocol option flags.
+    pub struct XLoadFlags: u16 {
+        /// This kernel has the legacy 64-bit entry point at `0x200`.
+        const XLF_KERNEL_64 = 1 << 0;
+        /// The kernel/boot_params/cmdline/ramdisk can be above 4G.
+        const XLF_CAN_BE_LOADED_ABOVE_4G = 1 << 1;
+        /// The kernel supports the 32-bit EFI handoff entry point given at [`SetupHeader::handover_offset`].
+        const XLF_EFI_HANDOVER_32 = 1 << 2;
+        /// The kernel supports the 64-bit EFI handoff entry point given at [`SetupHeader::handover_offset`] + `0x200`.
+        const XLF_EFI_HANDOVER_64 = 1 << 3;
+        /// The kernel supports kexec EFI boot with EFI runtime support.
+        const XLF_EFI_KEXEC = 1 << 4;
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Display, FromRepr)]
+#[repr(u32)]
+pub enum SetupDataType {
+    None = 0,
+    E820Ext = 1,
+    DTB = 2,
+    PCI = 3,
+    EFI = 4,
+    AppleProperties = 5,
+    Jailhouse = 6,
+    CCBlob = 7,
+    IMA = 8,
+    RngSeed = 9,
+}
+
+#[repr(C, packed)]
+pub struct SetupData {
+    pub next: *const SetupData,
+    pub type_: SetupDataType,
+    pub len: u32,
+    // We omit the data field here, as it can't be represented in safe Rust.
+}
+
+#[repr(C, packed)]
+pub struct CCSetupData {
+    pub header: SetupData,
+    pub cc_blob_address: u32,
+}
+
+/// Real-mode Kernel Header.
+///
+/// For each field, some are information from the kernel to the bootloader (“read”), some are
```

Comment:
```suggestion
/// For each field, some are information from the kernel to the bootloader ("read"), some are
```

maybe use standard double quotes, unless this was auto generated by bindgen, in which case feel free to leave as is.

Reply:
Nope, that's a manual copy-paste from the Linux kernel docs; I didn't even notice they were wrong. Fixed.